File: src/csharp/Intel/Iced.UnitTests/Intel/FormatterTests/SymbolResolverTestCases.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if GAS || INTEL || MASM || NASM || FAST_FMT
using System.Collections.Generic;
using System.Linq;

namespace Iced.UnitTests.Intel.FormatterTests {
	static class SymbolResolverTestCases {
		public static readonly (SymbolResolverTestCase[] testCases, HashSet<int> ignored) AllTests = GetTests();

		static (SymbolResolverTestCase[] testCases, HashSet<int> ignored) GetTests() {
			var filename = PathUtils.GetTestTextFilename("SymbolResolverTests.txt", "Formatter");
			var ignored = new HashSet<int>();
			return (SymbolResolverTestsReader.ReadFile(filename, ignored).ToArray(), ignored);
		}
	}
}
#endif

File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/fmt/SymbolResolverTestCases.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.fmt;

import java.util.HashSet;

import com.github.icedland.iced.x86.PathUtils;

final class SymbolResolverTestCases {
	public static class Info {
		public SymbolResolverTestCase[] tests;
		public HashSet<Integer> ignored;

		Info(SymbolResolverTestCase[] tests, HashSet<Integer> ignored) {
			this.tests = tests;
			this.ignored = ignored;
		}
	}

	public static final Info allTests = getTests();

	private static Info getTests() {
		String filename = PathUtils.getTestTextFilename("Formatter", "SymbolResolverTests.txt");
		HashSet<Integer> ignored = new HashSet<Integer>();
		return new Info(SymbolResolverTestsReader.readFile(filename, ignored).toArray(new SymbolResolverTestCase[0]), ignored);
	}
}
